### ISSUE TITLE
[Package Request] Pi-Hole

### ISSUE THREAD

**Issue by iroQuai** [SynoCommunity/spksrc#2501]
**Pi-Hole**
Description:  Network-wide hardware ad blocking on DNS level. Or: "A black hole for Internet advertisements (designed for Raspberry Pi). "

- Website: https://pi-hole.net/
- Software documentation: https://discourse.pi-hole.net/
- Build/Installation documentation: https://discourse.pi-hole.net/t/how-do-i-install-pi-hole-on-a-synology-nas/289
- Source code: https://github.com/pi-hole/pi-hole
- License: GNU GENERAL PUBLIC LICENSE (https://github.com/pi-hole/pi-hole/blob/master/LICENSE)

**Comment by flamme-demon** [comment]
Yes. Good idea

**Comment by iroQuai** [comment]
Especially interesting with the synology router!

**Comment by cytec** [comment]
Seems like this is more a collection of debian tools which are installed and configured to work properly when you run the installer... as there are A LOT of dependencies (ex: dnsmasq, lighthttpd, php5, nmap, cronie) and they use whiptail to show dialog boxes... i don't think this is something one would "just make a spk"...

Try installing it with debian chroot like stated on the website that should be a lot easyer!

**Comment by iroQuai** [comment]
Tnx for the reply! I had no idea. Your probably right then, but we'll see. For now I'll try the way explained on their site!

**Comment by stannynuytkens** [comment]
I'd really love this on my DS! I seems like the DNS Server package provided by Synology is based on [BIND9](http://bind9.net/). I'm not sure, it was so in the past. Can't find info, too lazy to SSH..

I don't think we'll need all the dependencies...
Basic functionality would only require:
- a source file with links to providers of adblock lists.
- a sheduler to update.
- a parser to parse the lists (duh) and basic front-end configuration.

This would of course only be possible if the generated "hosts" file can be used in combination of BIND9.
Or we could use dnsmasq? Somebody already ported that to a SPK. [http://syndnsmasq.the-ninth.com/](url)

Note, I have NO knowledge of creating a SPK, am only a Debian enthousiast, humble developer (non-python) but I do believe this is possible. Either as Pi-Hole or as a seperate slimmed-down package.

**Comment by cytec** [comment]
well i guess thats more like a slimmed-down pi-hole then.
Note: Installing Pi-Hole as it is should work in the debian chroot

**Comment by iroQuai** [comment]
It does, anf is documented quite good on the pi-hole discourse forums!

Op ma 13 mrt. 2017 19:31 schreef cytec <notifications@github.com>:

**Comment by stannynuytkens** [comment]
But I'm having issues getting the Debian chroot community package to run, it hangs on loading.. Maybe I'll try again later, if not I'll do a new pihole/squid(guard) setup.

**Comment by roelbroersma** [comment]
How do you guys fix the port-80 problem?   Synology is standard running (redirecting and running a reverse proxy) on port 80.

Pi_hole needs port 80.
I tried running the docker container and run into this problem.  The fix would be adding a second IP address to the Synology...   ?  someone?

**Comment by iroQuai** [comment]
That's all explained on the PiHole FAQ (check the link in the first post)

Copied that part for you:
The Pi-hole webserver runs on port 80, which is usually already occupied by
the webserver, so we need to modify the webserver config:

nano /etc/lighttpd/lighttpd.conf

modify server.port = xxxx to a suitable port (higher than 1023)
ctrl+x to save

Op ma 3 apr. 2017 15:16 schreef Roel Broersma <notifications@github.com>:

**Comment by roelbroersma** [comment]
Yes, i saw that,  But it is said that running the Pi-Hole webserver running on any other port than 80 is noot recommended and might give issues.

I can understand this because the webserver is serving things like:   spacer.gif
I am not sure how it behaves when you visit http(s)://www.cnn.com  and some spacer.gif comes from: http://192.168.x.x:1023/spacer.gif   (does this work without safety warning popups,..in all browsers..?)

This is just an example I can imagine,  but if I'm wrong or you have it working on another port and can confirm it's working fine,  please tell me.

**Comment by iroQuai** [comment]
No it doesn't work flawlessly, at least not for me. Replaced all the ads
with the synology logo (because that was the image found on the standard
webpage on port80). Didn't find any way to change it. Changed to using
PiHole on a regular pi (zero). Works like a charm now!

More info on why PiHole won't support custom ports here:
https://github.com/pi-hole/pi-hole/issues/1328

Op di 4 apr. 2017 12:50 schreef Roel Broersma <notifications@github.com>:

**Comment by ivar** [comment]
Another vote for a pihole package - it'd be awesome to make it easy to setup & use pihole on my NAS!

**Comment by jocamero** [comment]
+1 PiHole is great! I've been running it for years. It'd be great if it was easier for Synology users to install this.

**Comment by th0ma7** [comment]
I looked into creating a package but pihole is really meant to run in a docker container form and difficult for crossed-compiled environments such as Synology.  Not that this isn't feasible but would take a lot of efforts to get things going.

**Comment by mreid-tt** [comment]
I recently tried this in a Docker container using the [How to Setup Pi-hole on a Synology NAS](https://www.wundertech.net/how-to-setup-pi-hole-on-a-synology-nas-two-methods/) guide. I didn't do anything fancy and just used the first method and so far it seems to be working well. I'd much rather use a package than a Docker container but all in all, running on port 8888 it seems to be working like a champ. It's even using my custom external DNS providers and a handful of internal DNS names so that I can move these services off my router. So far I've seen two log warnings for high CPU in the app but I believe this happened when my Synology was doing disk scrubbing at the same time. No sustained performance issue or client slowdowns from what I can tell.
